packages/node/src/indexer/fetch.service.ts

```diff
@@ -271,28 +261,38 @@ export class FetchService implements OnApplicationShutdown {
     await this.getFinalizedBlockHead();
     await this.getBestBlockHead();
 
-    const validChecker = this.dictionaryValidation(
-      await this.dictionaryService.getSpecVersionsRaw(),
+    const rawSpecVersions = await this.dictionaryService.getSpecVersionsRaw();
+
+    const validChecker = this.dictionaryValidation(rawSpecVersions);
+
+    this.runtimeService.init(
+      this.getUseDictionary.bind(this),
+      this.getLatestFinalizedHeight.bind(this),
+      this.api,
     );
 
-    if (this.useDictionary && validChecker) {
-      const specVersionResponse =
-        await this.dictionaryService.getSpecVersions();
-      if (specVersionResponse !== undefined) {
-        this.specVersionMap = specVersionResponse;
-      }
+    if (validChecker) {
+      this.runtimeService.setSpecVersionMap(rawSpecVersions);
     } else {
-      this.specVersionMap = [];
+      this.runtimeService.setSpecVersionMap(undefined);
     }
 
     await this.blockDispatcher.init(
-      this.getRuntimeVersion.bind(this),
       this.resetForNewDs.bind(this),
+      this.runtimeService,
     );
 
     void this.startLoop(startHeight);
   }
 
+  getUseDictionary(): boolean {
+    return this.useDictionary;
+  }
+
+  getLatestFinalizedHeight(): number {
+    return this.latestFinalizedHeight;
+  }
```

Comment:
These seem rather redundant. What happens if we just provide the values directly to the service? I think they should update.